src/main/java/org/eln2/mc/common/blocks/cell/WireCellBlock.kt

```diff
@@ -1,11 +1,167 @@
 package org.eln2.mc.common.blocks.cell
 
+import net.minecraft.core.BlockPos
+import net.minecraft.core.Direction
 import net.minecraft.resources.ResourceLocation
+import net.minecraft.world.level.BlockGetter
+import net.minecraft.world.level.Level
+import net.minecraft.world.level.LevelReader
+import net.minecraft.world.level.block.Block
+import net.minecraft.world.level.block.HorizontalDirectionalBlock
+import net.minecraft.world.level.block.state.BlockState
+import net.minecraft.world.level.block.state.StateDefinition
+import net.minecraft.world.level.block.state.properties.DirectionProperty
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.phys.shapes.CollisionContext
+import net.minecraft.world.phys.shapes.VoxelShape
 import org.eln2.mc.common.blocks.CellBlockBase
+import org.eln2.mc.common.blocks.CellTileEntity
 import org.eln2.mc.common.cell.CellRegistry
 
 class WireCellBlock : CellBlockBase() {
+
+    lateinit var connect_east: BooleanProperty
+    lateinit var connect_west: BooleanProperty
+    lateinit var connect_north: BooleanProperty
+    lateinit var connect_south: BooleanProperty
+
+
+    init {
+
+        registerDefaultState(stateDefinition.any().setValue(connect_east, false).setValue(connect_north, false).setValue(connect_south, false).setValue(connect_west, false))
+    }
+
+
     override fun getCellProvider(): ResourceLocation {
         return CellRegistry.WIRE_CELL.id
     }
+
+    override fun getCollisionShape(pState: BlockState, pLevel: BlockGetter, pPos: BlockPos, pContext: CollisionContext): VoxelShape {
+        return Block.box(0.0, 0.0, 0.0, 16.0, 1.0, 16.0)
+    }
+
+    override fun getShape(pState: BlockState, pLevel: BlockGetter, pPos: BlockPos, pContext: CollisionContext): VoxelShape {
+        return Block.box(0.0, 0.0, 0.0, 16.0, 2.0, 16.0)
+    }
+
+    override fun onPlace(pState: BlockState, pLevel: Level, pPos: BlockPos, pOldState: BlockState, pIsMoving: Boolean) {
+        super.onPlace(pState, pLevel, pPos, pOldState, pIsMoving)
+
+        // get blocks around and call nechang
+        for (x in (-1 until 2)) {
+            for (z in (-1 until 2)) {
```

Comment:
I actually liked the logic you had before better. The until syntax makes this more confusing.

Reply:
Reversed the change.